wordpress-mobile/WordPress-iOS
# Tracks: Add site type details to media events

## Description

As a follow on from https://github.com/wordpress-mobile/WordPress-iOS/pull/22328, this PR adds site type details to media tracking events. The addition of these details will help us to further analyse these events in the future. 

More specifically, `is_jetpack`, `is_atomic`, and `is_hosted_at_WPcom` props have been added to the events, so that we can determine the type of site that the media events belonged to.

These new props have been added to the following events:

- `wpios_media_service_upload_started` **||** `jpios_media_service_upload_started`
- `wpios_media_service_upload_successful` **||** `jpios_media_service_upload_successful`
- `wpios_media_service_upload_canceled` **||** `jpios_media_service_upload_canceled`
- `wpios_media_service_upload_failed` **||** `jpios_media_service_upload_failed`


### Testing

* Install the build available in this PR's comments or checkout this branch locally.
* Navigate to the post editor and upload an image from your device's library.
* Before the image finished uploading, tap on it to cancel the process.
* Next, go through the steps to upload another image.
* Before the process finishes, turn off the WiFi so that the upload fails.
* Navigate to the **Live View** tab in Tracks.
* Enter the username associated with your testing account into the **Username, ID, or Email** field. 
* Wait until the events listed in this PR's **description** section show up in your history. _Note, this can take up to 20 minutes._
* When the events show up, verify that the new props are visible when expanding their details. The following table shows the value that the props should display depending on the site type you're using to test.

| Property  | Simple .com | Atomic | Jetpack | Self-hosted (without Jetpack) |
| ------------- | ------------- | ------------- | ------------- | ------------- |
| `is_atomic` | 0 | 1 | 0 | 0 |
| `is_hosted_at_wpcom ` | 1 | 0 | 0 | 0 |
| `is_jetpack` | 0 | 1 | 1 | 0 |

## Regression Notes
1. Potential unintended areas of impact

Any other changes to the way we collect data, beyond adding blog type details to the `*_media_service_upload_*` events, would be unintentional.

2. What I did to test those areas of impact (or what existing automated tests I relied on)

I considered whether it may be valuable to add site type details to all events it was possible for, but decided against this in order to avoid any unintended side effects, especially as I could not find a demand for this type of extra data for other events. By intentionally limited the changes to the `*_media_service_upload_*` events., I hope to avoid any unintended negative impact.

Aside from this, I compared the props for the `*_media_service_upload_*` events before this change and then after to verify there was nothing unexpected.

3. What automated tests I added (or what prevented me from doing so)

There are no existing tests in place for events, so I did not add any. As this PR only attaches a new piece of data to existing events, I did not feel a need to further investigate ways to automate testing for events in general.

<hr />

PR submission checklist:
